lib/daru/index.rb

```diff
@@ -81,7 +81,13 @@ def [](*key)
 
         slice first, last
       when key.size > 1
-        Daru::Index.new key.map { |k| self[k] }
+        if include? key[0]
+          Daru::Index.new key.map { |k| k }
+        else
+          # Assume the user is specifing values for index not keys
+          # Return index object having keys corresponding to values provided
+          Daru::Index.new key.map { |k| key k }
+        end
       else
```

Comment:
@v0dro I am on my way to improve the indexing infrastructure. I have made a major change in Index behavior. I think an example will illustrate it best:

```ruby
[3] pry(main)> i = Daru::Index.new([:a, :b, :c])
=> #<Daru::Index:0x00000003f5ca50
 @keys=[:a, :b, :c],
 @relation_hash={:a=>0, :b=>1, :c=>2},
 @size=3>
[4] pry(main)> i[0, 2]
=> #<Daru::Index:0x00000003ed14f0
 @keys=[:a, :c],
 @relation_hash={:a=>0, :c=>1},
 @size=2>
```

Earlier it used to be:

```ruby
[1] pry(main)> i = Daru::Index.new([:a, :b, :c])
=> #<Daru::Index:0x000000031f89b8
 @keys=[:a, :b, :c],
 @relation_hash={:a=>0, :b=>1, :c=>2},
 @size=3>
[2] pry(main)> i[0, 1]
=> #<Daru::Index:0x000000031bccd8
 @keys=[0, 1],
 @relation_hash={0=>0, 1=>1},
 @size=2>
```

The motive behind this is to remove the frequent checking of type of index and moving the functionality of guessing that the user is perhaps giving the index values (not keys) from `Vector` class to `Index` class because it seems more inherent to index. 

Are you good with this?

If you are good with this then I plan to the same with `MultiIndex` and finally remove the conditionals.